src/store.ts

```diff
@@ -413,32 +409,115 @@ export default function Store(
                 this.SDKConfig.onCreateBatch = undefined;
             }
         }
+    }
+}
 
-        if (!config.hasOwnProperty('flags')) {
-            this.SDKConfig.flags = {};
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            )
-        ) {
-            this.SDKConfig.flags[
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            ] = Constants.DefaultConfig.uploadInterval;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.ReportBatching
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.ReportBatching] = false;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.OfflineStorage
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.OfflineStorage] = 0;
+export interface IFeatureFlags {
+    reportBatching?: string;
+    eventBatchingIntervalMillis?: string;
+    offlineStorage?: string;
+    directURLRouting?: boolean;
+}
+
+export function processFlags(
```

Comment:
This might be a topic of conversation for a future revision, but I honestly think we should remove FeatureFlags and Endpoints from the Store and make them their own classes, or just move this processing logic outside of the file to make it smaller. Just leaving this comment as a reminder that we should chat about it.